doc/routing.md

```diff
@@ -0,0 +1,104 @@
+# Routing
+
+You can define your message handlers like every service in Symfony.
+But you need to tell your message bus that a handler is a target for a message object.
+
+There are two ways to route a message object to a handler.
+You can configure it using a tag at the handler definition or directly at the bus configuration.
+
+The basis for both ways is the service definition of the handler
+```yaml
+# app/config/services.yml
+services:
+    acme.command.register_user_handler:
+        class: Acme\Command\RegisterUserHandler
+```
+
+and a configured message bus:
+```yaml
+# app/config/config.yml or (flex) config/packages/prooph_service_bus.yml
+prooph_service_bus:
+    command_buses:
+        acme_command_bus: ~
+```
+The command bus is used as example, handlers are routed to each bus (nearly) the same way.
+
+## Routing using tags
+
+If you don't know about tags in Symfony please have a look at the
+[official documentation](http://symfony.com/doc/current/service_container/tags.html).
+
+To route a message to a specific handler, we just need to add a tag to its service definition:
+```yaml
+# app/config/services.yml
+services:
+    acme.command.register_user_handler:
+        class: Acme\Command\RegisterUserHandler
+        tags:
+            - { name: 'prooph_service_bus.acme_command_bus.route_target', message: Acme\Command\RegisterUser }
+```
+
+The name of the tag is simple `prooph_service_bus.<name-of-the-bus>.route_target`.
+The additional `message` attribute defines the class of the message object that is routed to the handler.
```

Comment:
It can also be a custom message name, so maybe: "message attribute defines the message name (message class by default) ..."